### DISCUSSION TITLE
Cara mengurutkan tanggal dalam plot (R studio)

### DISCUSSION THREAD

**Discussion by yohanespm77** [teamalgoritma/community#609] [Q&A] [answered]
Bagaimana cara mengurutkan tanggal dan bulan ketika ingin dimunculkan di plot ?

Script saya seperti ini : 
<img width="631" alt="image" src="https://user-images.githubusercontent.com/106010284/229680267-37168cf9-bea6-4f5d-8f0c-961553a10721.png">

Tapi ketika ditampilkan plotnya, tanggalnya tidak berurut : 
<img width="534" alt="image" src="https://user-images.githubusercontent.com/106010284/229680106-38c7eeb9-9e32-44e7-af96-47c1a0b73d3c.png">

Mohon bantuannya kak, terima kasih 🙏

**Comment by Yusraf002** [comment]
Dear @yohanespm77 

Terima kasih atas pertanyaannya pak, tim kami akan segera coba bantu atau carikan referensi ya pak

**Comment by IrfanChairurrachman** [comment]
Dear @yohanespm77 ,

Terima kasih telah mengirimkan kode, ekspektasi hasil dan permasalahan yg dialami di Github community ini.

Jika dibreakdown, tujuan yang ingin bapak capai adalah mengurutkan plot berdasarkan tanggalnya. Namun yang terjadi adalah plot Bapak tidak terurut berdasarkan tanggal. Hal ini terjadi karena x-axis yang bapak gunakan, yaitu `as.factor(day)` berupa faktor yang mengurutkan berdasarkan abjad (dalam hal ini angka). Sehingga, kami menyarankan untuk x-axisnya tetap menggunakan `tanggal` yang memiliki tipe data waktu (atau mengubah kembali `day` menjadi tipe data Date), tipe data date ketika dijadikan sebagai x-axis akan otomatis dibuat terurut.

Sekian jawaban dari kami, jika jawaban ini dirasa membantu bisa klik "Mark as Answer" pada comment ini.

Terima kasih,
Team Algoritma

**Reply by yohanespm77** [reply]
Saya coba bikin seperti ini kak : 
<img width="599" alt="image" src="https://user-images.githubusercontent.com/106010284/229729380-15ccc77b-380d-4ce0-8085-c96222551fb0.png">

Tapi tanggalnya tidak muncul semua 
<img width="625" alt="image" src="https://user-images.githubusercontent.com/106010284/229729479-6850d9d5-04f9-4d53-963f-a9a660b4b819.png">

Bagaimana agar tanggalnya muncul semua ?

**Reply by HeinzSiahaan** [reply, marked as answer]
Bisa dicoba untuk ditambahkan code berikut
`+ scale_x_date( date_breaks = "1 day" )`
